### PR TITLE
Disable flaky ogrprovider test

### PR DESCRIPTION
**Pull request by m-kuhn**
CC @troopa81 
https://github.com/qgis/QGIS/commit/e948120b7bf42b26eed432c2e2e0b43272c3b8fe#diff-92b7792b25e29c3b28e258ea893c309d

### CODE REVIEW

File: tests/src/providers/testqgsogrprovider.cpp [hunk 1]
```diff
@@ -184,6 +184,12 @@ void failOnWarning( QtMsgType type, const QMessageLogContext &context, const QSt
 
 void TestQgsOgrProvider::testThread()
 {
+  // Disabled by @m-kuhn
+  // This test is flaky
+  // See https://travis-ci.org/qgis/QGIS/jobs/505008602#L6464-L7108
+  return;
```
Comment (nyalldawson) [inline tests/src/providers/testqgsogrprovider.cpp]: Why not just disable on Travis? There's a utility function in the test library which detects this

Comment (m-kuhn) [reply tests/src/providers/testqgsogrprovider.cpp]: Thanks

**Comment by m-kuhn** [comment]
:jack_o_lantern:  one flaky test result masked by another one

**Comment by troopa81** [comment]
@m-kuhn This test was breaking from time to time? do you have a travis build failed, just curious!

**Comment by m-kuhn** [comment]
Yep, link in the code. https://github.com/qgis/QGIS/pull/9491/commits/8c676ba0b3d1fde74ee261c86a137d60b40c42cb#diff-92b7792b25e29c3b28e258ea893c309dR189

A timeout, probably a deadlock I guess.

**Comment by troopa81** [comment]
@m-kuhn Are you sure this is ``testThread`` that generates the timeout? 

Because I just launched the test on my machine and ``testThread`` works like a charm while ``testGpkgDataItemRename``fails and display a popup! This could explain the timeout.

Normally the different lock/wait in the test are not dependent from what happens in QGIS, it should not block the test even if it fails! But maybe I miss something

**Comment by m-kuhn** [comment]
@troopa81 no, I'm not completely sure.


My assumption is based on the two tests before being run successful, and I assumed this one would be executed after and never finishes https://travis-ci.org/qgis/QGIS/jobs/505008602#L6468-L6470.

I'm more than happy if you add some debug output and re-enable the test to see where it's blocked.

**Comment by troopa81** [comment]
It's done. Wait & see now!

**Pull request merged**